### PR TITLE
QSS Pack Contribution

### PR DESCRIPTION
**Pull request by QuantumSecSys**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/QSS/Integrations/QSS/QSS.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+- display: Fetch incidents
+  name: isFetch
+  required: false
+  type: 8
```
Comment (barchen1) [inline Packs/QSS/Integrations/QSS/QSS.yml:79]: Please remove.
duplicate field.

**Review by barchen1 — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution!
Here are some notes that you need to fix:
1. `demisto-sdk validate` failed cause you need to update the docker image in the yml file and also remove the duplicate `fetch parameter.
2. `demisto-sdk lint` failed cause your unitests failed, please fix them.
3. Please use `demisto-sdk generate-docs` to create a README file for the integration- https://xsoar.pan.dev/docs/documentation/readme_file#run-demisto-sdk-generate-docs.

File: Packs/QSS/Integrations/QSS/QSS.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+''' CONSTANTS '''
+
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+MAX_INCIDENTS_TO_FETCH = 50
+HELLOWORLD_SEVERITIES = ['Low', 'Medium', 'High', 'Critical']
```
Comment (barchen1) [inline Packs/QSS/Integrations/QSS/QSS.py:20]: Please change the name here

**Review by barchen1 — CHANGES_REQUESTED** [review]

File: Packs/QSS/Integrations/QSS/QSS.py (hunk 2 above)
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.py:20]: Done

File: Packs/QSS/Integrations/QSS/QSS.yml (hunk 1 above)
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.yml:79]: Done

File: Packs/QSS/Integrations/QSS/QSS.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+    for alert in alerts:
+
+        incident_created_time = int(alert.get('created_sec', '0'))
+
+        if last_fetch:
+            if incident_created_time <= last_fetch:
+                continue
+
+        incident_name = 'SOC Case ' + alert['reference']
```
[the remark this answers is not in the document]
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.py:154]: Done

File: Packs/QSS/IncidentFields/incidentfield-QSS_Case_Flase_Positive.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+    "associatedToAll": false,
+    "associatedTypes": [
+        "QSS SOC Monitoring"
+    ],
+    "caseInsensitive": true,
+    "cliName": "qsscaseflasepositive",
```
[the remark this answers is not in the document]
Comment (QuantumSecSys) [reply Packs/QSS/IncidentFields/incidentfield-QSS_Case_Flase_Positive.json:7]: Done

**Comment by barchen1** [comment]
We had a demo last week with Thamir, but the fetch incident didn't work as expected.
Thamir is working on a fix and we will have another demo next Monday.

File: Packs/QSS/pack_metadata.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "SOC Case Management Platform",
```
Comment (barchen1) [inline Packs/QSS/pack_metadata.json:2]: Please change the name to Quantum Security Systems

**Review by barchen1 — CHANGES_REQUESTED** [review]

File: Packs/QSS/Integrations/QSS/QSS.py [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+    if isinstance(arg, str):
+        if arg.isdigit():
+            return int(arg)
+        raise ValueError(f'Invalid number: "{arg_name}"="{arg}"')
+    if isinstance(arg, int):
+        return arg
+    raise ValueError(f'Invalid number: "{arg_name}"')
+
+
+def arg_to_timestamp(arg: Any, arg_name: str, required: bool = False) -> Optional[int]:
```
[the remark this answers is not in the document]
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.py:91]: Done

File: Packs/QSS/Integrations/QSS/QSS.py [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+
+    :return:
+    :rtype:
+    """
+
+    base_url = urljoin(demisto.params()['url'], '')
+
+    verify_certificate = not demisto.params().get('insecure', False)
+
+    first_fetch_time = arg_to_timestamp(
```
[the remark this answers is not in the document]
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.py:192]: Done

File: Packs/QSS/Integrations/QSS/QSS.py [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+            # This is the call made when pressing the integration Test button.
+            result = test_module(client, first_fetch_time)
+            return_results(result)
+
+        elif demisto.command() == 'fetch-incidents':
+
+            # Set and define the fetch incidents command to run after activated via integration settings.
+            alert_status = demisto.params().get('alert_status', None)
+            alert_type = demisto.params().get('alert_type', None)
+            min_severity = demisto.params().get('min_severity', None)
```
[the remark this answers is not in the document]
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.py:223]: Done

File: Packs/QSS/pack_metadata.json (hunk 5 above)
Comment (QuantumSecSys) [reply Packs/QSS/pack_metadata.json:2]: Done

File: Packs/QSS/Integrations/QSS/QSS.py [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+
+            # Convert the argument to an int using helper function or set to MAX_INCIDENTS_TO_FETCH
+            max_results = arg_to_int(
```
[the remark this answers is not in the document]
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.py:226]: Done

File: Packs/QSS/Integrations/QSS/QSS.py [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+def convert_to_demisto_severity(severity: str) -> int:
+    return {
+        'Low': 1,  # low severity
+        'Medium': 2,  # medium severity
+        'High': 3,  # high severity
+        'Critical': 4   # critical severity
+    }[severity]
+
+
+def arg_to_int(arg: Any, arg_name: str, required: bool = False) -> Optional[int]:
```
[the remark this answers is not in the document]
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.py:76]: Done

File: Packs/QSS/Integrations/QSS/QSS.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.py:192]: I tried to use `arg_to_datetime` but it shows the following error:

> Script failed to run: 
Error: [Traceback (most recent call last):
   File "<string>", line 243, in <module>
   File "<string>", line 171, in main
 NameError: name 'arg_to_datetime' is not defined
] (2604) (2603)

I put pack the old function `arg_to_timestamp`

File: Packs/QSS/Integrations/QSS/QSS.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (QuantumSecSys) [reply Packs/QSS/Integrations/QSS/QSS.py:76]: I tried but is it shows the following error:

> Error details
Script failed to run: 
Error: [Traceback (most recent call last):
   File "<string>", line 247, in <module>
   File "<string>", line 240, in main
   File "<string>", line 3218, in return_error
 Exception: Failed to execute fetch-incidents command.
Error:
name 'arg_to_number' is not defined
] (2604) (2603)

I kept the old function `arg_to_int`

**Comment by barchen1** [comment]
@QuantumSecSys Please fix the ["Run unit testing and lint"](https://app.circleci.com/pipelines/github/demisto/content/83892/workflows/7ea3b58c-c262-40b9-bbc5-a62d7100c066/jobs/346033) step so we can merge this PR

**Comment by barchen1** [comment]
@QuantumSecSys The `Run Unit Testing And Lint` stage is still failing, please fix it.

**Pull request merged**